client/client.go

```diff
@@ -349,11 +349,14 @@ func (c *simpleHTTPClient) Do(ctx context.Context, act httpAction) (*http.Respon
 
 	select {
 	case <-ctx.Done():
-		err = resp.Body.Close()
-		<-done
-		if err == nil {
-			err = ctx.Err()
+		if err = resp.Body.Close(); err != nil {
```

Comment:
You are right. It should focus on major error instead of cleanup error. That matches the expectation of users. I will change the logic.